theano/tensor/tests/test_slinalg.py

```diff
@@ -229,6 +229,45 @@ def test_solve_correctness(self):
         assert numpy.allclose(scipy.linalg.solve_triangular(U_val, b_val, lower=False),
                               upper_solve_func(U_val, b_val))
 
+    def test_solve_dtype(self):
```

Comment:
We should test integer to float too, that's where the out type inference is likely to break.